### PR TITLE
feat(service-portal): Health - Organ donation

### PR DESCRIPTION
**Pull request by disaerna**
# Health - Organ donation UI 🫁 

## What

UI for organ donation 

## Why
First phase of moving all "Heilsuvera" related data to  "Mínar síður"

## Screenshots / Gifs

Screenshot 2024-06-13 at 13 01 44
Screenshot 2024-06-13 at 12 56 41

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Added support for organ donation services in the health portal, enhancing user interactions with localized messages and navigation.
	- Introduced new components for organ donation management, including a registration form and limitations handling.
	- Expanded localization capabilities with new message entries to improve user experience related to organ donation.
	- Integrated new API scopes for enhanced functionality in health services.

- **Bug Fixes**
	- Improved error handling and data fetching for organ donation components to ensure a smoother user experience.

- **Documentation**
	- Updated documentation to reflect new API scopes related to health and organ donation.

### CODE REVIEW

File: libs/service-portal/health/src/module.tsx [hunk 1]
```diff
@@ -189,5 +197,19 @@ export const healthModule: PortalModule = {
       enabled: userInfo.scopes.includes(ApiScope.healthDentists),
       element: <DentistRegistration />,
     },
+    {
+      name: hm.organDonation,
+      path: HealthPaths.HealthOrganDonation,
+      key: 'HealthOrganDonation',
+      enabled: userInfo.scopes.includes(ApiScope.healthOrganDonation), // TODO: Change to correct scope!
```
Comment (thorkellmani) [inline libs/service-portal/health/src/module.tsx:204]: todo?

File: libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            >
+              <Button
+                variant="utility"
+                size="small"
+                icon="open"
+                iconType="outline"
+              >
+                {formatMessage(m.readAboutOrganDonation)}
+              </Button>
+            </LinkResolver>
```
Comment (thorkellmani) [inline libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx:39]: Doesn't the `<LinkButton /> `do this?

File: libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+              variant="eyebrow"
+              color="purple400"
+              marginTop={5}
+              marginBottom={1}
+            >
+              {formatMessage(m.takeOnOrganDonation)}
+            </Text>
+            <ActionCard
+              heading={data.data.title}
+              text={data.data.description}
```
Comment (thorkellmani) [inline libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx:52]: `data.data` 👀

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  const { data, loading, error } = getOptions(lang)
+
+  const [radioValue, setRadioValue] = useState<string | undefined>(undefined)
+
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const formData = new FormData(e.currentTarget)
+    const data = Object.fromEntries(formData.entries())
+    console.log(data)
```
Comment (thorkellmani) [inline libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:37]: 👀

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    // TODO: SEND DATA TO API
```
Comment (thorkellmani) [inline libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:38]: 👀

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    toast.success(formatMessage(messages.registrationComplete))
+    navigate(HealthPaths.HealthOrganDonation, { replace: true })
+
+    // on error
+    // toast.error(formatMessage(messages.registrationFailed))
+  }
```
Comment (thorkellmani) [inline libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:44]: commented out?

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                padding={3}
+                key={`organ-donation-${xi}`}
+              >
+                <RadioButton
+                  id={x.id}
+                  name="organ-registration-value"
+                  label={x.title}
+                  key={`organ-donation-${x.id}`}
+                  value={x.id}
+                  checked={radioValue === x.id} // TODO: Add default value as current user value
```
Comment (thorkellmani) [inline libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:73]: Todo!

**Review by thorkellmani — APPROVED** [review]
Cool, some todos

File: libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx (hunk 2 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx:39]: Ah might be, I didnt even think of it 🫨  I'll take a look

File: libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx (hunk 3 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx:52]: Just mock data here 😇

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx (hunk 4 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:37]: Will remove when connected to the api

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx (hunk 5 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:38]: Will remove when connected to the api

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx (hunk 6 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:44]: Will remove when connected to the api

File: libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx (hunk 7 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx:73]: Will be fixed when connected to the api

File: libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx (hunk 2 above)
Comment (disaerna) [reply libs/service-portal/health/src/screens/OrganDonation/OrganDonation.tsx:39]: Ill keep it like that, the button gets the whole width and i would have to add css class to fix

**Review by snaerseljan — APPROVED** [review]
LGTM!

**Pull request merged**